hoot-core/src/main/cpp/hoot/core/criterion/WayHeadingVarianceCriterion.h

```diff
@@ -50,15 +50,15 @@ class WayHeadingVarianceCriterion : public ElementCriterion, public ConstOsmMapC
   WayHeadingVarianceCriterion(
     const Degrees comparisonVariance, const NumericComparisonType& numericComparisonType,
     ConstOsmMapPtr map);
-  ~WayHeadingVarianceCriterion() = default;
+  ~WayHeadingVarianceCriterion() override = default;
 
   /**
    * @see ElementCriterion
    */
   bool isSatisfied(const ConstElementPtr& e) const override;
 
   ElementCriterionPtr clone() override
-  { return ElementCriterionPtr(new WayHeadingVarianceCriterion(_map)); }
+  { return std::make_shared<WayHeadingVarianceCriterion>(WayHeadingVarianceCriterion(_map)); }
```

Comment:
@bwitham I _*think*_ that you don't even have to have the second instance of `WayHeadingVarianceCriterion` in there when using `std::make_shared`.  I think the following should work:
```suggestion
  { return std::make_shared<WayHeadingVarianceCriterion>(_map); }
```

Reply:
Ok, thanks. I'll try that out here and with the other std::make_shared changes I recently made.